esp/src/src-react/hooks/workunit.ts

```diff
@@ -132,6 +132,29 @@ export function useWorkunitSourceFiles(wuid: string): [SourceFile[], Workunit, W
     const [sourceFiles, setSourceFiles] = React.useState<SourceFile[]>([]);
     const [count, inc] = useCounter();
 
+    // sorts the WU source files alphabetically by parent name, then name
+    // with children immediately following parents
+    const sortFiles = React.useCallback(files => {
+        const sortedFiles = [];
+        const parents = files.filter(item => item.__hpcc_parentName === "");
+        const children = files.filter(item => item.__hpcc_parentName !== "");
+
+        parents.sort((a, b) => a.Name.localeCompare(b.Name));
+        children.sort((a, b) => {
+            const indexA = parents.findIndex(parent => parent.Name === a.__hpcc_parentName);
+            const indexB = parents.findIndex(parent => parent.Name === b.__hpcc_parentName);
+            return indexA - indexB;
+        });
+
+        parents.forEach(parent => {
+            sortedFiles.push(parent);
+            const relatedChildren = children.filter(child => child.__hpcc_parentName === parent.Name);
+            sortedFiles.push(...relatedChildren);
+        });
+
+        return sortedFiles;
+    }, []);
```

Comment:
Will this work for nesting > 2 levels deep?
Is it not possible to do it within a single sort call?